bld/CLMBuildNamelist.pm

```diff
@@ -3987,10 +3988,37 @@ sub setup_logic_fates {
 
     if (&value_is_true( $nl_flags->{'use_fates'})  ) {
         add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'fates_paramfile', 'phys'=>$nl_flags->{'phys'});
-        my @list  = (  "fates_spitfire_mode", "use_fates_planthydro", "use_fates_ed_st3", "use_fates_ed_prescribed_phys", 
-                       "use_fates_inventory_init","use_fates_fixed_biogeog", "use_fates_logging","fates_parteh_mode", "use_fates_cohort_age_tracking" );
+        add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'use_fates_sp', 'use_fates'=>$nl_flags->{'use_fates'} );
+        if ( &value_is_true($nl->get_value('use_fates_sp')) ) {
+           $nl_flags->{'use_fates_sp'} = ".true.";
+        } else {
+           $nl_flags->{'use_fates_sp'} = ".false.";
+        }
+        my @list  = (  "fates_spitfire_mode", "use_fates_planthydro", "use_fates_ed_st3", "use_fates_ed_prescribed_phys",
+                       "use_fates_inventory_init","use_fates_fixed_biogeog","use_fates_nocomp",
+                       "use_fates_logging","fates_parteh_mode", "use_fates_cohort_age_tracking" );
         foreach my $var ( @list ) {
- 	  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, $var, 'use_fates'=>$nl_flags->{'use_fates'} );
+ 	  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, $var, 'use_fates'=>$nl_flags->{'use_fates'},
+                      'use_fates_sp'=>$nl_flags->{'use_fates_sp'} );
+        }
+        #
+        # For FATES SP mode make sure no-competetiion, and fixed-biogeography are also set
+        # And also check for other settings that can't be trigged on as well
+        #
+        my $var = "use_fates_sp";
+        if ( defined($nl->get_value($var))  ) {
+           if ( &value_is_true($nl->get_value($var)) ) {
+              my @list = ( "use_fates_nocomp", "use_fates_fixed_biogeog" );
+              foreach my $var ( @list ) {
+                 if ( ! &value_is_true($nl->get_value($var)) ) {
+                    $log->fatal_error("$var is required when FATES SP is on (use_fates_sp)" );
+                 }
+              }
+              # spit-fire can't be on with FATES SP mode is active
```

Comment:
This looks good. I did add a couple comments.

As a future change I also wonder if there aren't other things that shouldn't be checked as well. I'm guessing parteh_mode can't be turned on, nor logging, nor planthydro with FATES-SP. But, let me know if I'm wrong, or if there's anything else I'm missing. If so we can add it as an issue for a bit-for-bit change to make later.

Reply:
Hi @ekluzek, that's true that we wouldn't use FATES-SP and logging, but we would want to allow for FATES-SP and hydraulics, and I think I can imagine a potential use case with FATES-SP and different parteh modes.  So if you want to add that, then I think only add the blocker for logging.

Reply:
OK cool. Thanks for that clarification.

Reply:
Documented in #1489.